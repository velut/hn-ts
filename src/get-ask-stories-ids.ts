import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getAskStoriesIds` returns the list of up to 200 IDs of the "Ask HN" stories.
 *
 * @see {@link https://github.com/HackerNews/API#ask-show-and-job-stories}
 */
export async function getAskStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/askstories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
