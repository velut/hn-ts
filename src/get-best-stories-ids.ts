import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getBestStoriesIds` returns the list of up to 500 IDs of the best HN stories.
 *
 * @see {@link https://github.com/HackerNews/API#new-top-and-best-stories}
 */
export async function getBestStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/beststories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
