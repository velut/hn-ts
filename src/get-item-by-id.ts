import { getRawItemById } from "./get-raw-item-by-id";
import { Item } from "./item";
import { normalizeRawItem } from "./normalize-raw-item";

/**
 * `getItemById` returns the `Item` with the given ID.
 * If the item is not available, `getItemById` returns `undefined`.
 *
 * @see {@link Item}
 * @see {@link https://github.com/HackerNews/API#items}
 */
export async function getItemById({
  id,
}: {
  id: number;
}): Promise<Item | undefined> {
  const rawItem = await getRawItemById({ id });
  if (!rawItem) {
    return undefined;
  }

  return normalizeRawItem({ rawItem });
}
