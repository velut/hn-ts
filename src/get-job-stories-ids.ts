import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getJobStoriesIds` returns the list of up to 200 IDs of the job HN stories.
 *
 * @see {@link https://github.com/HackerNews/API#ask-show-and-job-stories}
 */
export async function getJobStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/jobstories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
