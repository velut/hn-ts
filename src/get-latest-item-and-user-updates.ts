import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";
import { Updated } from "./updated";

/**
 * `getLatestItemAndUserUpdates` returns the latest updates for items and users.
 *
 * @see {@link https://github.com/HackerNews/API#changed-items-and-profiles}
 */
export async function getLatestItemAndUserUpdates(): Promise<Updated> {
  const endpoint = `${hnApiBaseUrl}/updates.json`;
  const res = await fetch(endpoint);
  const updates = await res.json();
  return updates;
}
