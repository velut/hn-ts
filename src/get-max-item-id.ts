import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getMaxItemId` returns the ID of the most recent item available from the API.
 *
 * @see {@link https://github.com/HackerNews/API#max-item-id}
 */
export async function getMaxItemId(): Promise<number> {
  const endpoint = `${hnApiBaseUrl}/maxitem.json`;
  const res = await fetch(endpoint);
  const id = await res.json();
  return id;
}
