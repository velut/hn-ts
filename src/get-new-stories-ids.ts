import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getNewStoriesIds` returns the list of up to 500 IDs of the newer HN stories.
 *
 * @see {@link https://github.com/HackerNews/API#new-top-and-best-stories}
 */
export async function getNewStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/newstories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
