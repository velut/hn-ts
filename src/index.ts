/**
 * This package TODO:
 *
 * @packageDocumentation
 */

export * from "./get-ask-stories-ids";
export * from "./get-best-stories-ids";
export * from "./get-item-by-id";
export * from "./get-job-stories-ids";
export * from "./get-max-item-id";
export * from "./get-new-stories-ids";
export * from "./get-raw-item-by-id";
export * from "./get-raw-user-by-id";
export * from "./get-show-stories-ids";
export * from "./get-top-stories-ids";
export * from "./get-user-by-id";
export * from "./hn-api-base-url";
export * from "./item";
export * from "./item-type";
export * from "./raw-item";
export * from "./raw-item-type";
export * from "./raw-user";
export * from "./user";
