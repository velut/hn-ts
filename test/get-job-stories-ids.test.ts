/** @jest-environment setup-polly-jest/jest-environment-node */
// See https://netflix.github.io/pollyjs/#/test-frameworks/jest-jasmine?id=supported-test-runners

import NodeHttpAdapter from "@pollyjs/adapter-node-http";
import FSPersister from "@pollyjs/persister-fs";
import * as path from "path";
import { setupPolly } from "setup-polly-jest";
import { getJobStoriesIds } from "../src";

describe("getJobStoriesIds", () => {
  setupPolly({
    adapters: [NodeHttpAdapter],
    persister: FSPersister,
    persisterOptions: {
      fs: {
        recordingsDir: path.resolve(__dirname, "./__recordings__"),
      },
    },
    recordFailedRequests: true,
  });

  it("returns a list of IDs", async () => {
    expect.assertions(2);

    const ids = await getJobStoriesIds();

    expect(ids.length).toBeGreaterThanOrEqual(1);
    expect(ids).toMatchSnapshot();
  });
});
