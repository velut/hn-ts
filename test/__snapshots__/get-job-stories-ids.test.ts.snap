// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getJobStoriesIds returns a list of IDs 1`] = `
[
  35790672,
  35779079,
  35775687,
  35771604,
  35764445,
  35747058,
  35740226,
  35734516,
  35731138,
  35726744,
  35719928,
  35716811,
  35712321,
  35706760,
  35699275,
  35693229,
  35690256,
  35665701,
  35660076,
  35656726,
  35650563,
  35646253,
  35642891,
  35637505,
  35635858,
  35630550,
  35627071,
  35620078,
  35612819,
  35606563,
  35598625,
  35589553,
  35578305,
  35564291,
  35558539,
  35538271,
  35523895,
  35517364,
  35514470,
  35508903,
  35501918,
  35500585,
  35491192,
  35480687,
  35476736,
  35474365,
  35467186,
  35458644,
  35452614,
  35438223,
  35436068,
  35427338,
  35369529,
  35347760,
  35344225,
  35338138,
  35329153,
  35321609,
  35304593,
  35271343,
]
`;
